src/dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.cpp

```diff
@@ -258,7 +258,7 @@ void CXXNaiveCodeGen::generateStencilClasses(
       continue;
 
     // fields used in the stencil
-    const auto& StencilFields = stencil.getFields();
+    const auto& StencilFields = orderMap(stencil.getFields());
```

Comment:
no impact in performance, nor what the compiler will do with it, but the `&` could be confusing for readability, since the function is constructing and returning a new object, instead of returning simply a reference. 
Wouldnt `const auto stencilFields` be more clear? (notice also `s` instead of `S`)

Reply:
good, then that is all, if you remove the & in all these cases, and all testing in green go ahead with merge